### PR TITLE
Post-purchase experience: remove pending-migration sticker

### PR DESCRIPTION
**Pull request by andres-blanco**
## Proposed Changes

Based on this suggestion: p1730446326656409/1730331964.338199-slack-C0Q664T29  this diff stops the `migration-pending` sticker from being set in the `How do you want to migrate?` step.

## Why are these changes being made?

There's a bug that's been affecting users since this code was deployed. Having the sticker on a site where it doesn't belong will trigger usability issues. More details in the thread linked above.

## Testing Instructions

* Apply this PR
* Navigate through the migration flow until you get to the 
image

 * Verify that no request is being triggered to set the `migration-pending` sticker. This is the request that's being triggered now:
image

 * **Verify that navigating via both cards works as expected**

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by renatho — APPROVED** [review]
Looks good and works well!

We just need to fix the tests. I'm already approving, so after fixing the tests you can already merge it. 😉

I also added some other suggestions to the thread p1730476814238559/1730331964.338199-slack-C0Q664T29, but if we decide to do some of that it could be iterated later since it would be already an improvement for this situation.

**Review by valterlorran — CHANGES_REQUESTED** [review]
We need to update the tests to reflect the changes. One of the tests is expecting the sticker update to be called.

**Review by renatho — APPROVED** [review]
Looks good and works well!

Looking again, I thought that since it's a temporary solution, we could just tweak the `usePendingMigrationStatus` hook and comment the functions that call the API. For the tests, we could add a skip.

So we would avoid PR conflicts with a revert later. But if it's reverted very soon it should not be an issue either. So I'm already approving with the suggestion, and you can decide. 😉

**Comment by andres-blanco** [comment]
@renatho 
> So we would avoid PR conflicts with a revert later. But if it's reverted very soon it should not be an issue either. So I'm already approving with the suggestion, and you can decide. 😉

Probably I am missing something, but what PR conflicts will appear by removing the file?

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/test/index.tsx
Comment (valterlorran) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/test/index.tsx:113]: Shouldn't we keep these tests?

**Review by valterlorran — COMMENTED** [review]
I didn't test it, but changes are looking good. I was just not sure if we should remove all those tests.

**Review by gabrielcaires — APPROVED** [review]
I tested and it worked as expected with no regression. 

Regarding the test my suggestion is delete them, since  we will still need to find a solution that can completely different of this approach.

**Comment by gabrielcaires** [comment]
I can add back an updated version of the test file as part of a PR I will publish very soon.

**Pull request merged**
